### DISCUSSION TITLE
How to add this writing style  options.add_argument("--no-sandbox")

### DISCUSSION THREAD

**Discussion by luckyboy-wei** [seleniumbase/SeleniumBase#3363] [Q&A] [answered]
`I'm` not used to using native methods. This is the writing style I'm used to. How to add the highest permission when this browser starts，I only need to use methods with the highest permission “test_edit_scene” and call methods from other modules. If I use class inheritance methods, it will conflict with the drivers of other modules. I don't want to use class inheritance

options.add_argument("--no-sandbox")

`#用例级别标记
@pytest.mark.B
def test_new_scene(sb):
    dir_name = 'WEI测试专用'
    index_name = aa + uuid.uuid1().hex[:8]
    sb.open(conf_dict["home_url"])
    sb.maximize_window()
    sb.type('//input[@placeholder="请输入用户名"]', conf_dict["username"])
    sb.type('//input[@placeholder="请输入密码"]', conf_dict["password"])
    sb.click('//button/span')
    sb.assert_element_visible('//div[(text()="登录成功")]')
    return index_name

This method requires the highest authority to edit
#用例级别标记
@pytest.mark.B
def test_edit_scene(sb):
    infoname = test_new_scene(sb)
    sb.refresh()
    sb.sleep(2)
    sb.type('//input[@type="text"]',infoname)
`

**Comment by mdmintz** [comment, marked as answer]
`--no-sandbox` is already included by default:
https://github.com/seleniumbase/SeleniumBase/blob/05bb5657addf337944c1f954706ca1623e8cc559/seleniumbase/core/browser_launcher.py#L2224

For all formats, see: https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md

**Reply by luckyboy-wei** [reply]
The default permissions are not effective, and my online editing scene cannot be loaded. It must be rewritten to take effect and loaded   I run it under Windows




![PixPin_2024-12-24_11-23-53](https://github.com/user-attachments/assets/e46ffabf-a8d2-49a3-8642-2750304133fe)

`   
 class BaseClass(BaseCase):
        def get_new_driver(self, *args, **kwargs):
            options = webdriver.ChromeOptions()
            options.add_experimental_option(
                "excludeSwitches", ["enable-automation"]
            )
            prefs = {
                'credentials_enable_service': False,
                'profile.password_manager_enabled': False
            }
            options.add_experimental_option('prefs', prefs)
            options.add_experimental_option("useAutomationExtension", False)
            options.add_argument("--no-sandbox")
            options.add_argument("--disable-dev-shm-usage")
            options.add_argument("user-agent=YOUR_USER_AGENT")

            return webdriver.Chrome(options=options)
`


This usage doesn't work either


![image](https://github.com/user-attachments/assets/fd2b6928-452e-4b75-8557-0a38c985379d)

**Reply by mdmintz** [reply]
There's `chromium_arg="--no-sandbox"` if you want to try adding it.
There's also `enable_3d_apis=True` to enable the 3D APIs.

**Reply by luckyboy-wei** [reply]
Thank you
